### PR TITLE
Add checks for bad elements in DDL primary key

### PR DESCRIPTION
**Pull request by hazen**
- Make sure that the primary key has 3 elements
- Ensure that quantum is the last element
- Verify that the local key matches the same space as the primary key, i.e, same elements
- Properly report the errors back to `riak-admin`

### CODE REVIEW

**Comment by Basho-JIRA** [comment]
Issue 1244 of basho/riak_kv -  Add checks for bad elements in DLL primary key

- Make sure that the primary key has 3 elements 

***[posted via JIRA by Derek Somogyi]***

**Comment by Basho-JIRA** [comment]
Issue 1244 of basho/riak_kv -  Add checks for bad elements in DLL primary key

- Make sure that the primary key has 3 elements

https://github.com/basho/riak_kv/pull/1244 

***[posted via JIRA by Derek Somogyi]***

**Comment by Basho-JIRA** [comment]
Issue 1244 of basho/riak_kv -  Add checks for bad elements in DDL primary key
- Make sure that the primary key has 3 elements
- Ensure that quantum is the last element
- Verify that the local key matches the same space as the primary key, i.e, same elements
- Properly report the errors back to riak-admin

https://github.com/basho/riak_kv/pull/1244 

**_[posted via JIRA by Derek Somogyi]**_

File: src/riak_kv_wm_utils.erl [hunk 1]
```diff
@@ -497,6 +499,36 @@ assert_type_and_table_name_same(BucketType1, #ddl_v1{table = BucketType2}) ->
...
+%% @doc Verify that the first element of the primary key is a quantum
+assert_third_param_is_quantum(#hash_fn_v1{mod=riak_ql_quanta, fn=quantum}) ->
+    ok;
+assert_third_param_is_quantum(_KeyComponent) ->
+    throw({error, {primary_key, "Third element of primary key must be a quantum"}}).
+
+%% @doc Verify primary key and local partition have the same elements
+assert_primary_and_local_keys_match(#ddl_v1{partition_key = #key_v1{ast = Primary}, local_key = #key_v1{ast = Local}}) ->
+    PrimaryList = lists:sort([query_field_name(F) || F <- Primary]),
+    LocalList = lists:sort([query_field_name(F) || F <- Local]),
```
Comment (hmmr) [inline src/riak_kv_wm_utils.erl]: Isn't the order of the fields important, in both keys? Isn't equality of `PrimaryList` and `LocalList` a required and sufficient condition for the test condition to be met?

Comment (hazen) [inline src/riak_kv_wm_utils.erl]: Maybe you are correct.  I just needed to get the field names out.  I can take out the sort.

File: src/riak_kv_console.erl [hunk 2]
```diff
@@ -522,15 +522,13 @@ bucket_type_create(CreateTypeFn, Type, {struct, Fields}) ->
         [{<<"props", _/binary>>, {struct, Props1}}] ->
             case catch riak_kv_wm_utils:maybe_parse_table_def(Type, Props1) of
                 {ok, Props2} ->
+                    lager:info("OK Return"),
```
Comment (macintux) [inline src/riak_kv_console.erl]: Doesn't look like something we want in production, `info` or no

Comment (hazen) [inline src/riak_kv_console.erl]: Ooops.  Well spotted. @macintux

**Comment by hazen** [comment]
@borshop merge

**Pull request merged**